src/goto-programs/show_goto_functions_json.cpp

```diff
@@ -84,8 +84,11 @@ void show_goto_functions_jsont::show_goto_functions(
         instruction_entry["instructionId"]=
           json_stringt(instruction_id_builder.str());
 
-        instruction_entry["sourceLocation"]=
-          json(instruction.code.source_location());
+        if(instruction.code.source_location()!=source_locationt::nil())
```

Comment:
Is this any different from `instruction.code.source_location().is_not_nil()` ? - Really source_locationt::nil() should just go away, I'd say.